File: src/main/java/com/radixpro/enigma/xchg/domain/helpers/IndexMappingsList.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */

package com.radixpro.enigma.xchg.domain.helpers;

import java.util.List;

import static com.google.common.base.Preconditions.checkNotNull;

/**
 * Collection of all IndexMappings for a specific enum.
 */
public class IndexMappingsList {

   final List<IndexMapping> allIndexMappings;

   public IndexMappingsList(final List<IndexMapping> allIndexMappings) {
      this.allIndexMappings = checkNotNull(allIndexMappings);
   }

   public List<IndexMapping> getAllIndexMappings() {
      return allIndexMappings;
   }

   /**
    * Find sequenceId for a given enumId.
    *
    * @param enumId the enumId to be searched.
    * @return if found: the sequenceId, otherwise: -1 .
    */
   public int getSequenceIdForEnumId(final int enumId) {
      for (IndexMapping indexMapping : allIndexMappings) {
         if (indexMapping.getEnumId() == enumId) return indexMapping.getSequenceId();
      }
      return -1;
   }

   /**
    * Find enumId for a given sequenceId.
    *
    * @param sequenceId the sequenceId to be searched.
    * @return if found: the enumId, otherwise: -1.
    */
   public int getEnumIdForSequenceId(final int sequenceId) {
      for (IndexMapping indexMapping : allIndexMappings) {
         if (indexMapping.getSequenceId() == sequenceId) return indexMapping.getEnumId();
      }
      return -1;
   }
}

File: src/main/java/com/radixpro/enigma/xchg/domain/helpers/IndexMappingsList.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */
package com.radixpro.enigma.xchg.domain.helpers

/**
 * Collection of all IndexMappings for a specific enum.
 */
class IndexMappingsList(val allIndexMappings: List<IndexMapping>) {

    fun getSequenceIdForEnumId(enumId: Int): Int {
        for ((sequenceId, enumId1) in allIndexMappings) {
            if (enumId1 == enumId) return sequenceId
        }
        return -1
    }

    fun getEnumIdForSequenceId(sequenceId: Int): Int {
        for ((sequenceId1, enumId) in allIndexMappings) {
            if (sequenceId1 == sequenceId) return enumId
        }
        return -1
    }
}